### PR TITLE
Switch all components to functional ref (fix #970)

### LINKED ISSUE #970: Switch to functions for refs
See https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md

### PR DESCRIPTION
**Pull request by tofumatt**
There's the odd bit of reformatting/clean up in here too.

### CODE REVIEW

File: src/admin/components/SearchForm.js [hunk 1]
```diff
@@ -44,24 +44,25 @@ export class AdminSearchFormBase extends React.Component {
...
     return (
-      <form ref="form" className="search-form" method="GET"
-            action={pathname} onKeyDown={this.handleKeyDown}
+      <form method="GET" action={pathname} className="AdminSearchForm-form"
+            ref={(ref) => { this.form = ref; }} onKeyDown={this.handleKeyDown}
             onSubmit={(e) => e.preventDefault()}>
         <label className="visually-hidden" htmlFor="q">{_('Search')}</label>
-        <input ref="query" type="search" name="q" placeholder={_('Search')}
-               defaultValue={query} />
+        <input ref={(ref) => { this.query = ref; }} type="search" name="q"
```
Comment (muffinresearch) [inline src/admin/components/SearchForm.js]: This seems like it might be a bit too ambiguous in terms of naming? I wonder if we should use some kind of prefix for refs or other naming pattern to make it very clear what they are and to avoid the potential for collisions.

Comment (tofumatt) [inline src/admin/components/SearchForm.js]: Oh, fair enough. I don't know how to enforce that with the linter, but if we could I say yes.

Otherwise it's on us to enforce, which I guess works but is annoying.

Comment (tofumatt) [inline src/admin/components/SearchForm.js]: That said, actually, it's already more boilerplate than the string refs and that would just make it longer. The collisions would be pretty easy to spot (as they're all inside components). Maybe `query` itself is too ambiguous, but I'd rather just change this specific one to `inputQuery` but not have a generic `ref` prefix.

Comment (muffinresearch) [inline src/admin/components/SearchForm.js]: Yeah anything that gives a better hint as to what it represents without needing to the look at the code. That should be the yardstick.

File: src/core/css/SearchForm.scss [hunk 2]
```diff
@@ -1,6 +1,7 @@
 @import "~core/css/inc/vars";
 
-.search-form {
+.AdminSearchForm-form,
```
Comment (muffinresearch) [inline src/core/css/SearchForm.scss]: little sneaky unrelated clean-up ? :)

**Comment by muffinresearch** [comment]
r+wc

File: src/core/css/SearchForm.scss (hunk 2 above)
Comment (tofumatt) [inline src/core/css/SearchForm.scss]: Yeah, I added it by mistake but figured it wouldn't hurt much! :D
